File: TelerikUIExamples/TelerikUIExamples/SideDrawerHeaderView.m
//
//  SideDrawerHeaderView.m
//  TelerikUIExamples
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

#import "SideDrawerHeaderView.h"
#import <TelerikUI/TelerikUI.h>

@implementation SideDrawerHeaderView
{
    TKSideDrawerHeader *_sideDrawerHeader;
}

- (instancetype)initWithButton:(BOOL)addButton target:(id)target selector:(SEL)selector
{
    self = [self init];
    if (self) {
        _sideDrawerHeader = [[TKSideDrawerHeader alloc] initWithTitle:@"Navigation Menu"];
        _sideDrawerHeader.contentInsets = UIEdgeInsetsMake(0, 0, 0, 0);
        if (addButton) {
            UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
            [button setImage:[UIImage imageNamed:@"menu"] forState:UIControlStateNormal];
            [button addTarget:target action:selector forControlEvents:UIControlEventTouchUpInside];
            _sideDrawerHeader.actionButton = button;
            _sideDrawerHeader.contentInsets = UIEdgeInsetsMake(0, -20, 0, 0);
            _sideDrawerHeader.buttonPosition = TKSideDrawerHeaderButtonPositionLeft;
        }
        
        [self addSubview:_sideDrawerHeader];
    }
    
    return self;
}

- (void)layoutSubviews
{
    _sideDrawerHeader.frame = self.bounds;
}


@end

File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/SideDrawerHeaderView.swift
//
//  SideDrawerHeaderView.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

class SideDrawerHeaderView: UIView {
    
    let sideDrawerHeader = TKSideDrawerHeader(title: "Navigation Menu")
    
    convenience init (addButton: Bool, target: AnyObject?,selector: Selector?) {
        self.init()
        sideDrawerHeader.contentInsets = UIEdgeInsetsMake(-15, 0, 0, 0)
        if addButton {
            let button = UIButton(type:UIButtonType.System)
            button.setImage(UIImage(named: "menu"), forState: UIControlState.Normal)
            button.addTarget(target, action: selector!, forControlEvents: UIControlEvents.TouchUpInside)
            sideDrawerHeader.actionButton = button
            sideDrawerHeader.contentInsets = UIEdgeInsetsMake(-15, -20, 0, 0)
            sideDrawerHeader.buttonPosition = TKSideDrawerHeaderButtonPosition.Left
        }
        
        self.addSubview(sideDrawerHeader)
    }
    
    override func layoutSubviews() {
        sideDrawerHeader.frame = self.bounds
    }
}
